packages/contracts/contracts/ChugSplashManager.sol

```diff
@@ -116,12 +119,14 @@ contract ChugSplashManager is Owned {
     /**
      * @notice Address of the ChugSplashRegistry.
      */
-    ChugSplashRegistry public immutable registry;
+    // TODO: Add immutable keyword once this contract is not upgradeable anymore.
```

Comment:
Actually now that I'm thinking about it, we can make these immutable if we add a constructor back to this contract since the immutables will then live in the implementation. Just need a constructor that calls the `initialize` function.

Reply:
Unfortunately I don't think Solidity lets us do that. I tried to do this same thing when I was making the fixes for the OpenZeppelin audit, since they recommended adding the immutable keyword to variables in some implementation contracts (in L-14). However, this won't work because the variables are assigned outside of the constructor (in an initialize function). Adding the immutable keyword to these variables results in this error from the Solidity compiler: 
```TypeError: Cannot write to immutable here: Immutable variables can only be initialized inline or assigned directly in the constructor```.

Reply:
Ah I mean assign the immutables inside the constructor. Only use the constructor for the implementation contract.

Reply:
Yeah, we can do that to initialize the implementation contract, but how will the immutable variables be assigned in the proxy? We can't assign them in the initializer because immutables can only be assigned in the constructor.

Reply:
Ah the proxy gets the immutables from the implementation contract. I believe it should just work.

Reply:
I believe this must work or constants wouldn't work inside of implementations. Maybe worth checking briefly in Remix or something but 99% sure this should work out of the box.

Reply:
Ahh I see. Fixed!